greek-and-roman-musical-studies.csl

```diff
@@ -0,0 +1,291 @@
+        </group>
+      </if>
+      <else>
+        <text variable="edition" suffix=", "/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="additional-info">
+    <group prefix=" (" suffix=")" delimiter=", ">
+      <group>
+        <text variable="collection-title" suffix=" "/>
+        <text variable="collection-number"/>
+      </group>
+      <group>
+        <number variable="number-of-volumes" form="numeric"/>
+        <text term="volume" form="short" prefix=" " plural="true" strip-periods="true"/>
+      </group>
+      <text macro="edition"/>
+    </group>
+  </macro>
+  <macro name="publisher">
+    <group delimiter=": " prefix=" ">
+      <text variable="publisher-place"/>
+      <text variable="publisher"/>
+    </group>
+  </macro>
+  <macro name="year-date">
+    <choose>
+      <if variable="issued">
+        <date variable="issued">
+          <date-part name="year"/>
+        </date>
+      </if>
+      <else>
+        <text term="no date" form="short"/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="pages">
+    <text variable="page" prefix=" "/>
+  </macro>
+  <macro name="locator">
+    <choose>
+      <if locator="page">
+        <text variable="locator"/>
+      </if>
+      <else>
+        <group>
+          <label variable="locator" form="short"/>
+          <text variable="locator"/>
+        </group>
+      </else>
+    </choose>
+  </macro>
+  <macro name="doi">
+    <choose>
+      <if match="any" variable="DOI">
+        <group delimiter=" " prefix=". ">
+          <text value="DOI:"/>
+          <text variable="DOI"/>
+        </group>
+      </if>
+    </choose>
+  </macro>
+  <citation et-al-min="3" et-al-use-first="1" disambiguate-add-year-suffix="true" disambiguate-add-names="true" disambiguate-add-givenname="true">
+    <layout delimiter="; ">
+      <group delimiter=", ">
+        <group delimiter=" ">
+          <text macro="author-short"/>
+          <text macro="year-date" prefix="" suffix=""/>
+        </group>
+        <text macro="locator"/>
+      </group>
+    </layout>
+  </citation>
+  <bibliography et-al-min="4" et-al-use-first="1" hanging-indent="true">
+    <sort>
+      <key macro="author"/>
+      <key variable="issued"/>
+    </sort>
+    <layout suffix=".">
+      <text macro="author" font-variant="normal"/>
+      <date variable="issued" prefix=" (" suffix=").">
+        <date-part name="year"/>
+      </date>
+      <choose>
+        <if type="bill book graphic legal_case legislation motion_picture report song" match="any">
+          <group delimiter=". " prefix=" ">
+            <group prefix=" ">
+              <text macro="title" quotes="false"/>
+              <number variable="volume" form="numeric" prefix=" " suffix=" "/>
+              <text macro="editor"/>
+              <text macro="translator"/>
+              <text macro="additional-info"/>
+              <choose>
+                <if type="motion_picture" match="any" variable="URL">
```

Comment:
Since the `access` macro already contains a test for URL, this test is unnecessary